### PR TITLE
Custom Machine Config Pool role must match the MCP name

### PR DESCRIPTION
**Pull request by mburke5678**
If the user creates a MachineConfigPool with a name different than the `spec.machineConfigSelector.matchExpressions.machineconfiguration.openshift.io/role` a configuration issue with the KubeletConfig like https://issues.redhat.com/browse/OCPBUGS-10682 can happen

https://issues.redhat.com/browse/OCPBUGS-29784

Previews:
Creating a ContainerRuntimeConfig CR to edit CRI-O parameters -> [Example ContainerRuntimeConfig CR](https://73709--ocpdocs-pr.netlify.app/openshift-enterprise/latest/post_installation_configuration/machine-configuration-tasks.html#create-a-containerruntimeconfig_post-install-machine-configuration-tasks) -- Callout 1
Creating a KubeletConfig CRD to edit kubelet parameters -> [Consider the following guidance ](https://73709--ocpdocs-pr.netlify.app/openshift-enterprise/latest/post_installation_configuration/machine-configuration-tasks.html#create-a-kubeletconfig-crd-to-edit-kubelet-parameters_post-install-machine-configuration-tasks)-> Note after 3rd bullet. 

QE review:
- [ ] QE has approved this change.

### CODE REVIEW

File: modules/nodes-nodes-resources-configuring-auto.adoc [hunk 1]
```diff
@@ -74,7 +74,7 @@ spec:
 ----
 <1> Assign a name to CR.
 <2> Add the `autoSizingReserved` parameter set to `true` to allow {product-title} to automatically determine and allocate the `system-reserved` resources on the nodes associated with the specified label. To disable automatic allocation on those nodes, set this parameter to `false`.
-<3> Specify the label from the machine config pool that you configured in the "Prerequisites" section. You can choose any desired labels for the machine config pool, such as `custom-kubelet: small-pods`, or the default label, `pools.operator.machineconfiguration.openshift.io/worker: ""`.
```
Comment (yuqi-zhang) [inline modules/nodes-nodes-resources-configuring-auto.adoc:77]: I think this selector is unrelated, so what is here still applies as before.

This is a label selecting a kubeletconfig for a pool, whereas the issue in the description happens for labels selection machineconfigs for a pool (which is not the same thing)

File: modules/create-a-kubeletconfig-crd-to-edit-kubelet-parameters.adoc [hunk 2]
```diff
@@ -16,12 +16,32 @@ As the fields in the `kubeletConfig` object are passed directly to the kubelet f
...
-
 * Edit an existing `KubeletConfig` CR to modify existing settings or add new settings, instead of creating a CR for each change. It is recommended that you create a CR only to modify a different machine config pool, or for changes that are intended to be temporary, so that you can revert the changes.
 
+* Create one `KubeletConfig` CR for each machine config pool with all the config changes you want for that pool. If you are applying the same content to all of the pools, you need only one `KubeletConfig` CR for all of the pools.
+
 * As needed, create multiple `KubeletConfig` CRs with a limit of 10 per cluster. For the first `KubeletConfig` CR, the Machine Config Operator (MCO) creates a machine config appended with `kubelet`. With each subsequent CR, the controller creates another `kubelet` machine config with a numeric suffix. For example, if you have a `kubelet` machine config with a `-2` suffix, the next `kubelet` machine config is appended with `-3`.
 
+[NOTE]
+====
+If you are applying a kubelet or container runtime config to a custom machine config pool, the custom role in the `machineConfigSelector` must match the name of the custom machine config pool.
```
Comment (yuqi-zhang) [inline modules/create-a-kubeletconfig-crd-to-edit-kubelet-parameters.adoc:27]: Do you think we should elaborate on why this is the case? I guess my concern is that this is how the MCP is structured, but doesn't really explain how to set up the kubeletconfig/containerruntimeconfig itself, so it might confuse readers.

the content itself is correct

File: modules/create-a-kubeletconfig-crd-to-edit-kubelet-parameters.adoc [hunk 3]
```diff
@@ -16,12 +16,32 @@ As the fields in the `kubeletConfig` object are passed directly to the kubelet f
 
 Consider the following guidance:
 
-* Create one `KubeletConfig` CR for each machine config pool with all the config changes you want for that pool. If you are applying the same content to all of the pools, you need only one `KubeletConfig` CR for all of the pools.
-
 * Edit an existing `KubeletConfig` CR to modify existing settings or add new settings, instead of creating a CR for each change. It is recommended that you create a CR only to modify a different machine config pool, or for changes that are intended to be temporary, so that you can revert the changes.
 
+* Create one `KubeletConfig` CR for each machine config pool with all the config changes you want for that pool. If you are applying the same content to all of the pools, you need only one `KubeletConfig` CR for all of the pools.
```
Comment (yuqi-zhang) [inline modules/create-a-kubeletconfig-crd-to-edit-kubelet-parameters.adoc:21]: So I know this is just a change of ordering but I think this may be incorrect from before. You should have a separate kubeletconfig for each pool, or there could be unintended errors.

File: modules/create-a-kubeletconfig-crd-to-edit-kubelet-parameters.adoc (hunk 2 above)
Comment (mburke5678) [reply modules/create-a-kubeletconfig-crd-to-edit-kubelet-parameters.adoc:27]: @yuqi-zhang If you have some context that will help the user understand this limitation, I would be happy to put it in.

**Comment by mburke5678** [comment]
@rioliu-rh Are you the right QE resource to review this PR?

File: modules/create-a-kubeletconfig-crd-to-edit-kubelet-parameters.adoc (hunk 2 above)
Comment (yuqi-zhang) [reply modules/create-a-kubeletconfig-crd-to-edit-kubelet-parameters.adoc:27]: Sorry, I missed your comment. I was thinking of something like:

This is generally not the case, but for kubelet and containerruntime config objects, the generated machineconfig that lays down the actual configuration always references the role as the generated machineconfig role. Thus if they do not match, the MCO will still generate the machineconfig but the MCP will not match it by default.

I'm not sure if that's too verbose though and could be even more confusing

File: modules/create-a-kubeletconfig-crd-to-edit-kubelet-parameters.adoc [hunk 4]
```diff
@@ -16,12 +16,32 @@ As the fields in the `kubeletConfig` object are passed directly to the kubelet f
...
+[source,yaml]
+----
+apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfigPool
+metadata:
+  name: infra
+spec:
+  machineConfigSelector:
+    matchExpressions:
+      - {key: machineconfiguration.openshift.io/role, operator: In, values: [infra]}
```
Comment (rioliu-rh) [inline modules/create-a-kubeletconfig-crd-to-edit-kubelet-parameters.adoc:40]:
```suggestion
      - {key: machineconfiguration.openshift.io/role, operator: In, values: [worker,infra]}
```

File: modules/create-a-kubeletconfig-crd-to-edit-kubelet-parameters.adoc [hunk 5]
```diff
@@ -16,12 +16,32 @@ As the fields in the `kubeletConfig` object are passed directly to the kubelet f
...
 
+* Create one `KubeletConfig` CR for each machine config pool with all the config changes you want for that pool.
+
 * As needed, create multiple `KubeletConfig` CRs with a limit of 10 per cluster. For the first `KubeletConfig` CR, the Machine Config Operator (MCO) creates a machine config appended with `kubelet`. With each subsequent CR, the controller creates another `kubelet` machine config with a numeric suffix. For example, if you have a `kubelet` machine config with a `-2` suffix, the next `kubelet` machine config is appended with `-3`.
 
+[NOTE]
+====
+If you are applying a kubelet or container runtime config to a custom machine config pool, the custom role in the `machineConfigSelector` must match the name of the custom machine config pool. 
+
+For example, because this custom machine config pool is named `infra`, the custom role must also be `infra`.
```
Comment (skopacz1) [inline modules/create-a-kubeletconfig-crd-to-edit-kubelet-parameters.adoc:29]: Suggestion - I think using a colon would more clearly relate this sentence to the YAML snippet below:
```suggestion
For example, because this custom machine config pool is named `infra`, the custom role must also be `infra`:
```

**Review by skopacz1 — COMMENTED** [review]
Just one suggestion, otherwise LGTM!
